delphi-checks/src/main/java/au/com/integradev/delphi/checks/InheritedMethodWithNoCodeCheck.java

```diff
@@ -195,6 +195,7 @@ private static boolean argumentSignaturesMatch(
   private static boolean isOverriddenMethod(
       RoutineNameDeclaration parent, RoutineNameDeclaration child) {
     return parent.getName().equalsIgnoreCase(child.getName())
-        && parent.getParameters().equals(child.getParameters());
+        && parent.getParameters().equals(child.getParameters())
+        && parent.isCallable();
```

Comment:
Class methods have polymorphism in Delphi and can call `inherited` too.
```suggestion
        && parent.isClassInvocable() == child.isClassInvocable()
        && parent.isCallable();
```

Reply:
Unfortunately, `inherited` on an instance routine can refer to a class routine in the parent:

```pascal
type
  TBase = class(TObject)
  protected
    class procedure MyProc;
  end;

  TChild = class(TBase)
  public
    procedure MyProc;
  end;

procedure TChild.MyProc;
begin
  Writeln('instance proc');
  inherited MyProc;
end;

class procedure TBase.MyProc;
begin
  Writeln('class proc');
end;

initialization
  TChild.Create.MyProc; // writes 'instance proc\nclass proc'
```

Reply:
I've implemented this logic and added some tests to confirm.